pkg/netpol/connlist/connlist.go

```diff
@@ -501,6 +505,12 @@ func (ca *ConnlistAnalyzer) getConnectionsBetweenPeers(pe *eval.PolicyEngine, pe
 			if err != nil {
 				return nil, nil, err
 			}
+			if ca.exposureAnalysis {
+				err = updatePeersGeneralExposureData(pe, srcPeer, dstPeer, ingressSet, egressSet, exposuresMap)
```

Comment:
can just call `addPeerGeneralExposure` on `src` peer for both ingress and egress (right after `src` is assigned), and then no need to have `ingressSet` and `egressSet`

Reply:
we can not update the general exposure before calling `pe.AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer)` at least once; 
since those fields are updated on the fly during the computation of the allowed conns between the src and dst.(on the first time a src appears in this call: its egress fields are updated; and in the first call with the dst: its ingress fields updated)
and therefore we can not update both egress and ingress peer's fields at once (we need a pod to have both "attitudes" in order to be updated) ; thus we need both egress and ingress sets.